atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/LocalWriteBuffer.java

```diff
@@ -118,8 +118,12 @@ private Map<Cell, ChangeMetadata> getChangeMetadataForTableInternal(TableReferen
         return metadataByTable.computeIfAbsent(tableRef, unused -> new HashMap<>());
     }
 
-    public long getByteCount() {
-        return byteCount.get();
+    public long getValuesByteCount() {
+        return valuesByteCount.get();
+    }
+
+    public long changeMetadataCount() {
+        return metadataByTable.values().stream().mapToLong(Map::size).sum();
     }
 
```

Comment:
we could also keep a byte count for metadata, but - like in TimeLock - we can use the count to compute an upper bound on the memory usage anyways.